src/main/java/org/ayty/hatcher/api/v1/competence/exception/RestExceptionHandler.java

```diff
@@ -0,0 +1,45 @@
+package org.ayty.hatcher.api.v1.competence.exception;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+@RestControllerAdvice
+public class RestExceptionHandler {
+
+	@ExceptionHandler(MethodArgumentNotValidException.class)
+	public ResponseEntity handleFieldValidation(MethodArgumentNotValidException ex) {
+		
+		List<String> errorsList = ex.getBindingResult()
+                .getAllErrors()
+                .stream()
```

Comment:
acho que quando você usar o ResponseStatus na exeption, não seja necessário essa implementação.

Reply:
Essa implementação é pra pegar a lista de mensagens de erros da exceção e construir um objeto personalizado pra exibir o erro como json, dessa forma: 
![image](https://user-images.githubusercontent.com/38544710/107992841-507d8100-6fb8-11eb-81a3-715580d8ed28.png)
Recomenda a implementação de outra forma? Dessa forma que tu falou tá parecendo que tu me recomendou usar o tratamento usando ResponseStatusExceptionResolver, só que tratando dessa forma, segundo esse texto da Baeldung: https://www.baeldung.com/exception-handling-for-rest-with-spring#3-responsestatusexceptionresolver  não é possível devolver um json no corpo da resposta, o que pode dificultar o tratamento no front.

Reply:
Ver com os outros devs a implementação das respostas de exceção, essa parece ser uma forma legal, mas foi implementado de jeito diferente dos demais, não vai dar muito certo quando for para a develop. Acertando com eles vocês definem quem vai enviar o tratamento de exceções